tests/vmi_headless_test.go

```diff
@@ -0,0 +1,159 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017, 2018 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	"flag"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	kubev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	v1 "kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/tests"
+)
+
+var _ = Describe("[rfe_id:609]VMIheadless", func() {
+
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+
+	var vmi *v1.VirtualMachineInstance
+
+	BeforeEach(func() {
+		tests.BeforeTestCleanup()
+		vmi = tests.NewRandomVMIWithEphemeralDisk(tests.ContainerDiskFor(tests.ContainerDiskAlpine))
+	})
+
+	Describe("[rfe_id:609]Creating a VirtualMachineInstance", func() {
+
+		Context("with headless", func() {
+
+			BeforeEach(func() {
+				f := false
+				vmi.Spec.Domain.Devices.AutoattachGraphicsDevice = &f
+			})
+
+			It("should create headless vmi without any issue", func() {
+				tests.RunVMIAndExpectLaunch(vmi, 30)
+			})
+
+			It("[test_id:714][posneg:positive]should not have vnc graphic device in xml", func() {
+				tests.RunVMIAndExpectLaunch(vmi, 30)
+
+				runningVMISpec, err := tests.GetRunningVMISpec(vmi)
+				Expect(err).ToNot(HaveOccurred(), "should get vmi spec without problem")
+
+				Expect(len(runningVMISpec.Devices.Graphics)).To(Equal(0), "should not have any graphics devices present")
+			})
+
+			It("[test_id:737][posneg:positive]should match memory with overcommit enabled", func() {
+				vmi.Spec.Domain.Resources = v1.ResourceRequirements{
+					Requests: kubev1.ResourceList{
+						kubev1.ResourceMemory: resource.MustParse("100M"),
+					},
+					OvercommitGuestOverhead: true,
+				}
+				vmi = tests.RunVMIAndExpectLaunch(vmi, 30)
+
+				readyPod := tests.GetPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+				computeContainer := tests.GetComputeContainerOfPod(readyPod)
+
+				Expect(computeContainer.Resources.Requests.Memory().String()).To(Equal("100M"))
+			})
+
+			It("[posneg:negative]should not match memory with overcommit disabled", func() {
```

Comment:
@petrkotas 
In this case you have vm set with Headless and over-commit is disabled.
It is like VM with Headless only that we check the memory (https://polarion.engineering.redhat.com/polarion/redirect/project/CNV/workitem?id=CNV-713) no?

Reply:
Yes, exactly.

Reply:
Use [CNV-2444](https://polarion.engineering.redhat.com/polarion/redirect/project/CNV/workitem?id=CNV-2444) for the negative test case.